src/main/java/uk/gov/companieshouse/api/accounts/service/impl/DirectorsApprovalService.java

```diff
@@ -0,0 +1,167 @@
+    private DirectorsReportServiceImpl directorsReportService;
+    private KeyIdGenerator keyIdGenerator;
+
+
+    @Autowired
+    public DirectorsApprovalService(DirectorsApprovalTransformer transformer, DirectorsApprovalRepository directorsApprovalRepository, DirectorsReportServiceImpl directorsReportService, KeyIdGenerator keyIdGenerator) {
+        this.transformer = transformer;
+        this.directorsApprovalRepository = directorsApprovalRepository;
+        this.directorsReportService = directorsReportService;
+        this.keyIdGenerator = keyIdGenerator;
+    }
+
+    @Override
+    public ResponseObject<DirectorsApproval> create(DirectorsApproval rest, Transaction transaction, String companyAccountId, HttpServletRequest request) throws DataException {
+
+        String directorApprovalID = generateID(companyAccountId);
+
+        setMetadataOnRestObject(rest, transaction, companyAccountId);
+
+        DirectorsApprovalEntity entity = transformer.transform(rest);
+
+        entity.setId(directorApprovalID);
+
+        try {
+
+            directorsApprovalRepository.insert(entity);
+        } catch (DuplicateKeyException dke) {
+
+            return new ResponseObject<>(ResponseStatus.DUPLICATE_KEY_ERROR);
+        } catch (MongoException ex) {
+
+            throw new DataException(ex);
+        }
+
+        directorsReportService.addLink(companyAccountId, DirectorsReportLinkType.APPROVAL, getSelfLink(rest), request);
+
+        return new ResponseObject<>(ResponseStatus.CREATED, rest);
+    }
+
+    @Override
+    public ResponseObject<DirectorsApproval> update(DirectorsApproval rest, Transaction transaction, String companyAccountId, HttpServletRequest request) throws DataException {
+
+        String directorApprovalID = generateID(companyAccountId);
+
+        setMetadataOnRestObject(rest, transaction, companyAccountId);
+
+        DirectorsApprovalEntity entity = transformer.transform(rest);
+        entity.setId(directorApprovalID);
+
+        try {
+
+            directorsApprovalRepository.save(entity);
+        } catch (MongoException e) {
+
+            throw new DataException(e);
+        }
+        return new ResponseObject<>(ResponseStatus.UPDATED, rest);
+    }
+
+    @Override
+    public ResponseObject<DirectorsApproval> find(String companyAccountsId, HttpServletRequest request) throws DataException {
+
+        DirectorsApprovalEntity directorsApprovalEntity;
+
+        try {
+
+            directorsApprovalEntity =  directorsApprovalRepository.findById(generateID(companyAccountsId)).orElse(null);
+        } catch (MongoException e) {
+
+            throw new DataException(e);
+        }
+
+        if (directorsApprovalEntity == null) {
+            return new ResponseObject<>(ResponseStatus.NOT_FOUND);
+        }
+
+        DirectorsApproval directorsApproval = transformer.transform(directorsApprovalEntity);
+        return new ResponseObject<>(ResponseStatus.FOUND, directorsApproval);
```

Comment:
You could use 
```
return new ResponseObject<>(ResponseStatus.FOUND, transformer.transform(directorsApprovalEntity));
```
here to remove the needless variable assignment

Reply:
Removed the variable as suggested